core/src/main/java/fj/data/LazyString.java

```diff
@@ -108,7 +105,9 @@ public CharSequence subSequence(final int start, final int end) {
    * @return The String representation of this lazy string.
    */
   public String toString() {
-    return new StringBuilder(length() + 16).append(this).toString();
+    final StringBuilder builder = new StringBuilder(length() + 16);
+    s.foreachDoEffect(c -> builder.append(c.charValue()));
+    return builder.toString();
   }

```

Comment:
Because `StringBuilder.append(CharSequence)` calls `CharSequence.charAt`, but `LazyString.chatAt` calls `Sequence.index` which is O(n). As a result, `LazyString.toString` is O(n^2). Try to convert LazyString of 10000 characters - it will not finish in reasonable time.